Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -188,24 +214,29 @@ void UPSWorldSubsystem::OnGameStateChanged(ECurrentGameState CurrentGameState)
 // Load game from save file or create a new one (does initial load from data table)
 void UPSWorldSubsystem::LoadGameFromSave()
 {
+	// load from data table
+	const UDataTable* ProgressionDataTable = UPSDataAsset::Get().GetProgressionDataTable();
+	checkf(ProgressionDataTable, TEXT("ERROR: 'ProgressionDataTableInternal' is null"));
+	UMyDataTable::GetRows(*ProgressionDataTable, ProgressionSettingsDataInternal);
+
 	// Check if the save game file exists
-	if (UGameplayStatics::DoesSaveGameExist(SaveGameInstanceInternal->GetSaveSlotName(), SaveGameInstanceInternal->GetSaveSlotIndex()))
+	if (UGameplayStatics::DoesSaveGameExist(SaveGameDataInternal->GetSaveSlotName(), SaveGameDataInternal->GetSaveSlotIndex()))
 	{
-		SaveGameInstanceInternal = Cast<UPSSaveGameData>(UGameplayStatics::LoadGameFromSlot(SaveGameInstanceInternal->GetSaveSlotName(), SaveGameInstanceInternal->GetSaveSlotIndex()));
+		SaveGameDataInternal = Cast<UPSSaveGameData>(UGameplayStatics::LoadGameFromSlot(SaveGameDataInternal->GetSaveSlotName(), SaveGameDataInternal->GetSaveSlotIndex()));
 	}
 	else
 	{
 		// Save file does not exist
 		// do initial load from data table
-		TMap<FName, FPSRowData> SavedProgressionRows;
-		const UDataTable* ProgressionDataTable = UPSDataAsset::Get().GetProgressionDataTable();
-		checkf(ProgressionDataTable, TEXT("ERROR: 'ProgressionDataTableInternal' is null"));
-		UMyDataTable::GetRows(*ProgressionDataTable, SavedProgressionRows);
-		SaveGameInstanceInternal = Cast<UPSSaveGameData>(UGameplayStatics::CreateSaveGameObject(UPSSaveGameData::StaticClass()));
 
-		if (SaveGameInstanceInternal)
+		SaveGameDataInternal = Cast<UPSSaveGameData>(UGameplayStatics::CreateSaveGameObject(UPSSaveGameData::StaticClass()));
+
+		if (SaveGameDataInternal)
 		{
-			SaveGameInstanceInternal->SetProgressionMap(SavedProgressionRows);
+			for (auto Row : ProgressionSettingsDataInternal)
```

Comment:
Unreal Engine guidelines don't recommend `auto` usages
https://dev.epicgames.com/documentation/en-us/unreal-engine/epic-cplusplus-coding-standard-for-unreal-engine#auto
![image](https://github.com/user-attachments/assets/4150ed14-c339-4ada-bf0c-79a034ac7b59)

**Please update all your `auto` usages, I see few places**

### Expected result
```cppp
for (const TTuple<FName, FPSRowData> Row : ProgressionSettingsDataInternal)
```

Reply:
goti it, done: https://github.com/h4rdmol/ProgressionSystem/pull/18/commits/6cc0edaae7d031a88922fdaf33f35090c391a9f7